hyperspy/io.py

```diff
@@ -848,3 +852,55 @@ def save(filename, signal, overwrite=None, **kwds):
                 signal.tmp_parameters.set_item('folder', file.parent)
                 signal.tmp_parameters.set_item('filename', file.stem)
                 signal.tmp_parameters.set_item('extension', extension)
+
+
+def add_file_load_metadata(signal_dict, reader):
+    mdata_dict = {
+        'operation': 'load',
+        'io_plugin': reader.__loader__.name,
+        'hyperspy_version': hs_version,
+        'timestamp': datetime.now().astimezone().isoformat()
+    }
+    if 'General' not in signal_dict['metadata']:
+        # if we don't even have basic metadata, just return as-is (shouldn't
+        # happen, but this came up in certain tests)
+        return signal_dict
```

Comment:
this 'odd case' is not tested for (codecov warning)

Reply:
Locally, it was hit during the tests of the nexus IO plugin: https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/tests/io/test_nexus_hdf.py#L335

I'm not familiar with the Nexus reader, so I don't know why it tries to store metadata directly as a `dict` rather than a `DictionaryTreeBrowser` (that seems inconsistent), and caused an error at this part of my changes. Regardless, I could add a specific test for it, I suppose.

Reply:
Huh, now I see the same thing locally... not sure what that is. I'll just remove it if apparently it's not necessary.